### ISSUE TITLE
Repository tagui_local.csv with only one row fails - to be fixed in v6

### ISSUE THREAD

**Issue by siowyisheng** [kelaberetiv/TagUI#663]
Using an object repository with only one row fails. Once I add a newline after the header row, it works.

On Windows 10, this fails:

```
// flow
`website`
```

```
// tagui_local.csv
OBJECT,DEFINITION
website,https://github.com/kelaberetiv/TagUI
```

But this works
```
// tagui_local.csv
OBJECT,DEFINITION

website,https://github.com/kelaberetiv/TagUI
```

**Comment by kensoh** [comment]
Nice spotting this! I can replicate on macOS.

**Comment by kensoh** [comment]
Doing a print_r() in that section in tagui_parse.php (without empty 1st row) -
```
Array
(
    [0] => Array
        (
            [0] => website
            [1] => https://github.com/kelaberetiv/TagUI
        )

    [1] => 
)
```

Doing a print_r() in that section in tagui_parse.php (with empty 1st row) -
```
Array
(
    [0] => Array
        (
            [0] => 
        )

    [1] => Array
        (
            [0] => website
            [1] => https://github.com/kelaberetiv/TagUI
        )

    [2] => 
)
```

**Comment by kensoh** [comment]
Printing `$repo_count` shows that without empty line the count is 0 while with empty line the count is 1. Something is wrong here...

**Comment by kensoh** [comment]
I think the bug is in `$repo_count-=1;`  in both the tagui_local.csv and tagui_global.csv sections. Header row is discarded at the start, but there is still the -1 at the end. Thus for a 1-line csv file the repo_count ends up to be 0. Checking further..

**Comment by kensoh** [comment]
In function parse_backticks(), there is the following condition -

```php
    if ($GLOBALS['repo_count'] == 0) {
        $script_line = parse_variables($script_line);
    } else {
```

It doesn't try to search in repo_data if the count is 0. That means if header row is discarded and count is 0 for the 1 line of data, it would not be checked. Something has to be changed there or the way the header row is dropped.

**Comment by kensoh** [comment]
Hi @siowyisheng I think there is a working solution -

In tagui_parse.php there are 2 occurrences of below line, for tagui_local.csv and tagui_glocal.csv -
```php
if ($repo_count != 0) $repo_count++; fgetcsv($local_repo_file); // +1 if array has data, discard header record
```

The solution would be changing to below, discarding header row only if the array is populated -
```php
if ($repo_count != 0) {$repo_count++; fgetcsv($local_repo_file);} // +1 if array has data, discard header record
```
&ensp;
This way $repo_count will be accurate in reflecting if there is repository or data table data, and having $repo_count 0 means no data. I think even if no discarding of header row things should work fine, but the array in memory will be full of headers (eg 1 from tagui_local.csv, 1 from tagui_global.csv, and maybe 1 from data table). I think that's ugly engineering even if no user can see lol.

**Comment by kensoh** [comment]
It's been a while so I can't recall if discarding header rows always in above line is by design or a mistake forgetting to put brackets when I added above feature. You can do some sanity checks if above works to put in a PR for v6.

**Comment by kensoh** [comment]
I tried testing following cases and work, but have not tested with data tables.
- 1 row in tagui_local.csv, no tagui_global.csv
- 2 rows in tagui_local.csv, no tagui_global.csv
- no tagui_local.csv, 1 row tagui_global.csv
- no tagui_local.csv, 2 rows tagui_global.csv
- 1 row in tagui_local.csv, 1 row in tagui_global.csv
- 2 rows in tagui_local.csv, 2 rows in tagui_global.csv
- 1 row in tagui_local.csv, 1 row in tagui_global.csv, tested with modified samples/6A_datatables

**Comment by kensoh** [comment]
raised PR to merge into develop branch, unassigning myself

**Comment by siowyisheng** [comment]
fixed by @kensoh in #672 for v6 release
